csrc/host_ir/executor.cpp

```diff
@@ -14,6 +14,32 @@ namespace nvfuser {
 
 namespace hir {
 
+namespace {
+
+at::Tensor getKnownTensorOrEmptyTensor(
```

Comment:
```suggestion
at::Tensor getKnownTensorOrUndefined(
```

Naming nit: The word "empty" may confuse readers to think it's returned by `torch.empty`. In fact, `at::Tensor` creates an undefined tensor (i.e., `tensor.defined() == false`), whereas `torch.empty` creates an uninitialized but defined (with a valid shape/stride etc) tensor.